src/Sylius/Bundle/ResourceBundle/spec/Controller/ParametersParserSpec.php

```diff
@@ -25,165 +24,113 @@
-
-        $this->parseRequestValues(
-            [
-                'criteria' => '$criteria',
-                'sortable' => '$sorting',
-            ],
-            $request
-        )->shouldReturn(
-            [
-                'criteria' => 'New criteria',
-                'sortable' => 'New sorting',
-            ]
-        );
+        $request = new Request();
+        $request->request->set('array', ['foo' => 'bar']);
+
+        $this->beConstructedWith(new Container(), new ExpressionLanguage());
+
+        $this
+            ->parseRequestValues(['nested' => ['array' => '$array']], $request)
+            ->shouldReturn(['nested' => ['array' => ['foo' => 'bar']]])
+        ;
     }
 
-    function it_should_parse_complex_parameters(Request $request)
+    function it_should_parse_expressions()
     {
-        $request->get('enable')->willReturn(true);
-        $request->get('sorting')->willReturn('New sorting');
-
-        $this->parseRequestValues(
-            [
-                'criteria' => [
-                    'enable' => '$enable',
-                ],
-                'sortable' => '$sorting',
-            ],
-            $request
-        )->shouldReturn(
-            [
-                'criteria' => [
-                    'enable' => true,
-                ],
-                'sortable' => 'New sorting',
-            ]
-        );
+        $service = new \stdClass();
+
+        $container = new Container();
+        $container->set('service', $service);
+
+        $expression = new ExpressionLanguage();
+
+        $request = new Request();
+
+        $this->beConstructedWith($container, $expression);
+
+        $this
+            ->parseRequestValues(['nested' => ['service' => 'expr:service("service")']], $request)
+            ->shouldReturn(['nested' => ['service' => $service]])
+        ;
     }
 
-    function it_should_parse_expression_with_parameters(
-        ContainerInterface $container,
-        ExpressionLanguage $expression,
-        Request $request
-    ) {
-        $request->get('foo')->willReturn('bar');
-        $request->get('baz')->willReturn(1);
-
-        $expression->evaluate('service("demo_service")', ['container' => $container])->willReturn('demo_object');
-
-        $this->parseRequestValues(
-            [
-                'factory' => [
-                    'method' => 'createByParameter',
-                    'arguments' => [
-                        'expr:service("demo_service")',
-                    ],
-                ],
-            ],
-            $request
-        )->shouldReturn(
-            [
-                'factory' => [
-                    'method' => 'createByParameter',
-                    'arguments' => [
-                        'demo_object',
-                    ],
-                ],
-            ]
-        );
-
-        $expression
-            ->evaluate('service("demo_service")->getWith("bar")', ['container' => $container])
```

Comment:
Ideal example of "don't mock what you don't own" principle usefulness: `->` does not exist in expression language, `.` should be used instead :)